terraform/dotnet/k8s/deploy/main.tf

```diff
@@ -0,0 +1,169 @@
+# ------------------------------------------------------------------------
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+resource "null_resource" "deploy" {
+  connection {
+    type        = "ssh"
+    user        = var.user
+    private_key = var.ssh_key
+    host        = var.host
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      <<-EOF
+
+      # Make the Terraform fail if any step throws an error
+      set -e
+
+      # Ensure environment is clean
+      echo "LOG: Rerunning cleanup commands in case of cleanup failure in previous run"
+      helm uninstall --debug --namespace amazon-cloudwatch amazon-cloudwatch-operator --ignore-not-found
+      kubectl delete namespace dotnet-sample-app-namespace --ignore-not-found=true
+      [ ! -e helm-charts ] || sudo rm -r helm-charts
+      [ ! -e dotnet-frontend-service-depl.yaml ] || rm dotnet-frontend-service-depl.yaml
+      [ ! -e dotnet-remote-service-depl.yaml ] || rm dotnet-remote-service-depl.yaml
+
+      # Clone and install operator onto cluster
+      echo "LOG: Cloning helm charts repo"
+      git clone https://github.com/aws-observability/helm-charts.git -q
+      cd helm-charts/charts/amazon-cloudwatch-observability
+      git reset --hard d92e2850f6b88b5e47f4071e557f87b88d73fa75
```

Comment:
Let's update this based on #226 since it's merged and functional

Reply:
It's not for dotnet, just tested. Have discussed with Vastin to use this latest release version for now

Reply:
Done